presentation/src/main/java/me/androidbox/presentation/login/viewmodel/RegisterViewModel.kt

```diff
@@ -2,14 +2,26 @@ package me.androidbox.presentation.login.viewmodel
 
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.usecase.RegisterUseCase
+import me.androidbox.presentation.NetworkResponseState
+import retrofit2.HttpException
 import javax.inject.Inject
 
 @HiltViewModel
 class RegisterViewModel @Inject constructor(
-    private val savedStateHandle: SavedStateHandle
+    private val savedStateHandle: SavedStateHandle,
+    private val registerUseCase: RegisterUseCase
 ): ViewModel() {
 
+    private val registerMutableState = MutableStateFlow<NetworkResponseState<Unit>>(NetworkResponseState.Idle)
```

Comment:
Instead of an idle state for enums/sealed classes, I'd just make it nullable if it doesn't exist. That's exactly what nulls are used for.